### PR TITLE
Update script to get PSVersion from $PSVersionTable

### PR DESCRIPTION
**Pull request by daxian-dbw**
The file `powershell.version` has been removed as we now figure out `PSVersion` and `GitCommitId` from the `ProductVersion` of `S,M.A.dll`. Update the script to get the PSVersion directly from `$PSVersionTable.PSVersion`.

### CODE REVIEW

File: src/powershell-native/Install-PowerShellRemoting.ps1 [hunk 1]
```diff
@@ -10,15 +10,14 @@
...
-    [ValidateNotNullOrEmpty()]
+    [parameter(Mandatory = $true, ParameterSetName = "ByPath")]
     [string]
     $PowerShellHome,
 
-    [parameter(ParameterSetName = "ByPath")]
-    [ValidateNotNullOrEmpty()]
+    [parameter(Mandatory = $true, ParameterSetName = "ByPath")]
     [string]
     $PowerShellVersion = "6.0.0-alpha.8"
```
Comment (SteveL-MSFT) [inline src/powershell-native/Install-PowerShellRemoting.ps1]: If the path is provided, we should get the powershell version using `powershell -v`and get rid of this parameter

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Comment by iSazonov** [comment]
I wonder why the script is in `src/powershell-native` directory?

File: src/powershell-native/Install-PowerShellRemoting.ps1 (hunk 1 above)
Comment (adityapatwardhan) [reply src/powershell-native/Install-PowerShellRemoting.ps1]: I agree.

**Review by adityapatwardhan — APPROVED** [review]
Approved with minor comments.

File: src/powershell-native/Install-PowerShellRemoting.ps1 (hunk 1 above)
Comment (daxian-dbw) [reply src/powershell-native/Install-PowerShellRemoting.ps1]: Sounds good. Will make the change.

**Comment by daxian-dbw** [comment]
@iSazonov This script is to register WinRM endpoint configuration to make `pwrshplugin.dll` work for powershell core remoting. It's bundled together with pwrshplugin.dll in psrp.windows nuget package.

**Comment by mirichmo** [comment]
A few clean up items since we are touching this file:
1. Line 6 should be removed. It is no longer relevant.
1. The comment on line 39 should be cleaned up to be "powershell.6.0.0-beta.3"
1. Line 78 should be uncommented. The registry file doesn't get cleaned up during MSI uninstallation
1. The comment on line 116 should be moved to line 3 to become the introductory description since it provides an overview of the script. Right now, it is buried in the middle and hard to find.

File: src/powershell-native/Install-PowerShellRemoting.ps1 [hunk 2]
```diff
@@ -69,9 +73,6 @@ function Register-WinRmPlugin
 
     Write-Verbose "Performing WinRM registration with: $fileName"
     reg.exe import .\$fileName
-
-    # Clean up
-#    Remove-Item .\$fileName
```
Comment (mirichmo) [inline src/powershell-native/Install-PowerShellRemoting.ps1]: Clean up should still take place. Just remove the "#" on line 74

**Review by mirichmo — CHANGES_REQUESTED** [review]

File: src/powershell-native/Install-PowerShellRemoting.ps1 (hunk 2 above)
Comment (daxian-dbw) [reply src/powershell-native/Install-PowerShellRemoting.ps1]: Added that line back.

**Review by mirichmo — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@SteveL-MSFT Your comment has been addressed. Can you please take another look?
@adityapatwardhan The failure of macOS is not related to this PR. #5065 will fix macOS build.

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

**Review by adityapatwardhan — APPROVED** [review]
LGTM

**Comment by adityapatwardhan** [comment]
Restarted MacOS build.

**Comment by daxian-dbw** [comment]
@adityapatwardhan The build is still failing is it's not related to this PR.

**Pull request merged**